ou_dedetai/config.py

```diff
@@ -1,209 +1,1052 @@
+        """Returns absolute path to the wine binary"""
+        output = self._raw.wine_binary
+        if output is None:
+            question = f"Which Wine AppImage or binary should the script use to install {self.faithlife_product} v{self.faithlife_product_version} in {self.install_dir}?: "  # noqa: E501
+            options = utils.get_wine_options(self.app)
+
+            choice = self.app.ask(question, options)
+
+            output = choice
+            self.wine_binary = choice
+        # Return the full path so we the callee doesn't need to think about it
+        if self._raw.wine_binary is not None and not Path(self._raw.wine_binary).exists() and (Path(self.install_dir) / self._raw.wine_binary).exists(): # noqa: E501
+            return str(Path(self.install_dir) / self._raw.wine_binary)
+        return output
+
+    @wine_binary.setter
+    def wine_binary(self, value: str):
+        """Takes in a path to the wine binary and stores it as relative for storage"""
+        # Make the path absolute for comparison
+        relative = self._relative_from_install_dir(value)
+        # FIXME: consider this, it doesn't work at present as the wine_binary may be an
+        # appimage that hasn't been downloaded yet
+        # aboslute = self._absolute_from_install_dir(value)
+        # if not Path(aboslute).is_file():
+        #     raise ValueError("Wine Binary path must be a valid file")
+
+        if self._raw.wine_binary != relative:
+            self._raw.wine_binary = relative
+            # Reset dependents
+            self._raw.wine_binary_code = None
+            self._overrides.wine_appimage_path = None
+            self._write()
+
+    @property
+    def wine_binary_files(self) -> list[str]:
+        if self._wine_binary_files is None:
+            self._wine_binary_files = utils.find_wine_binary_files(
+                self.app, self.faithlife_product_release
+            )
+        return self._wine_binary_files
+
+    @property
+    def wine_app_image_files(self) -> list[str]:
+        if self._wine_appimage_files is None:
+            self._wine_appimage_files = utils.find_appimage_files(self.app)
+        return self._wine_appimage_files
+
+    @property
+    def wine_binary_code(self) -> str:
+        """Wine binary code.
+        
+        One of: Recommended, AppImage, System, Proton, PlayOnLinux, Custom"""
+        if self._raw.wine_binary_code is None:
+            self._raw.wine_binary_code = utils.get_winebin_code_and_desc(self.app, self.wine_binary)[0]  # noqa: E501
+            self._write()
+        return self._raw.wine_binary_code
+
+    @property
+    def wine64_binary(self) -> str:
+        return str(Path(self.wine_binary).parent / 'wine64')
+    
+    @property
+    # This used to be called WINESERVER_EXE
+    def wineserver_binary(self) -> str:
+        return str(Path(self.wine_binary).parent / 'wineserver')
+
+    # FIXME: seems like the logic around wine appimages can be simplified
```

Comment:
Possibly. These are handled differently in part because of how they are validated and we've noticed that appimage wine bottles and non-appimage wine bottles don't place nicely if you swap the binaries around.